iceplume.jl

```diff
@@ -227,7 +227,7 @@ end
 #@inline sponge_b(x, y, z, b, p) = -west_mask_cos(x, y, z) * p.σ * (b -  b_west(y, z, t, p)) -east_mask_cos(x, y, z) * p.σ * (b -  b_east(y, z, t, p))# nudges S to S∞
 
 @inline sponge_b(x, y, z , b, p) = -west_mask_cos(x, y, z) * p.σ * (b -  b_west(y, z, p)) -east_mask_cos(x, y, z) * p.σ * (b -  b_east(y, z, p))
```

Comment:
Taking a quick glace, I think your signature is wrong. For a 3D simulation, forcing functions can only have two signatures:

```julia
sponge_b(x, y, z, t) = .... # If you're not passing any parameters
```

or

```julia
sponge_b(x, y, z, t, p) = .... # If you're passing parameters, which need to be passing through a single variable
# Here I'm calling that vairiable `p` but ofc it can be called anything you want
```

That might not be the only problem, but it's definitely _a_ problem that needs fixing. Let me know if the error message continues after you fix this.

Reply:
I tried a long time to fix this, but I couldn't. I need to pass "b" to sponge_b in the same way I need to pass T to sponge_T, don't I?

Reply:
Ah, crap, I hadn't noticed `b` there was buoyancy. That's why you (meaning I) shouldn't do things trying to be fast. I was right in catching that you weren't passing `t`, and you should, but you also gotta pass `b`, which I missed in your code. Anyway, when in doubt of these things, we should always look to the docs. In this case this is what it says for `Forcing`:

```julia
help?> Forcing
search: Forcing AdvectiveForcing FluxBoundaryCondition FieldBoundaryConditions ConformalCubedSphereGrid

  Forcing(func; parameters=nothing, field_dependencies=(), discrete_form=false)

  Return a Forcing function, which can be added to the tendency of a model field.

  If discrete_form=false (the default), and neither parameters nor field_dependencies are provided, then func must be callable with the signature

  func(x, y, z, t)

  where x, y, z are the east-west, north-south, and vertical spatial coordinates, and t is time. Note that this form is also default in the constructor for
  NonhydrostaticModel, so that Forcing is not needed.

  If discrete_form=false (the default), and field_dependencies are provided, the signature of func must include them. For example, if field_dependencies=(:u, :S) (and
  parameters are not provided), then func must be callable with the signature

  func(x, y, z, t, u, S)

  where u is assumed to be the u-velocity component, and S is a tracer. Note that any field which does not have the name u, v, or w is assumed to be a tracer and must be
  present in model.tracers.

  If discrete_form=false (the default) and parameters are provided, then the last argument to func must be parameters. For example, if func has no field_dependencies but
  does depend on parameters, then it must be callable with the signature

  func(x, y, z, t, parameters)

  The object parameters is arbitrary in principle, however GPU compilation can place constraints on typeof(parameters).

  With field_dependencies=(:u, :v, :w, :c) and parameters, then func must be callable with the signature

  func(x, y, z, t, u, v, w, c, parameters)
```

The last example there is what's relevant for you. To use your names and variables (if I'm not forgetting anything), that means that you gotta write:

```julia
@inline sponge_b(x, y, z, t , b, p) = -west_mask_cos(x, y, z) * p.σ * (b -  b_west(y, z, p)) -east_mask_cos(x, y, z) * p.σ * (b -  b_east(y, z, p))
Fb = Forcing(sponge_b, field_dependencies = :b, parameters = params)
```

Can you please see if that works?